ci: cross-compile to iOS for PR

Bogdanp opened the pull request:
This ensures that cross-compilation to iOS for Racket CS works on PR. To test that things work at runtime, we could set up a free account at one of the iOS CI services and have it run daily builds. I'll look into that next weekend.

The PR workflows seemed like a convenient place to put this, but I'm happy to move it elsewhere if there's a better place (cc @pmatos).

pmatos commented on racket/src/ChezScheme/c/alloc.c:
```diff
@@ -284,9 +284,6 @@ void S_close_off_thread_local_segment(ptr tc, ISPC s, IGEN g) {
   close_off_segment(tgc, tgc->next_loc[g][s], tgc->base_loc[g][s], tgc->sweep_loc[g][s], s, g);
 
   tgc->base_loc[g][s] = (ptr)0;
-#if defined(WRITE_XOR_EXECUTE_CODE)
```
Shouldn't this be part of another PR? I would like to isolate CI patches, instead of having them together with changes that potentially break code.

pmatos commented on .github/workflows/ci-pr.yml:
```diff
@@ -66,3 +66,35 @@ jobs:
+        ../configure \
+          --host=aarch64-apple-darwin \
+          --enable-ios=iPhoneOS \
+          --enable-racket=${{ github.workspace}}/host-racket/bin/racket \
```
space after `github.workspace` for consistency.

pmatos commented on .github/workflows/ci-pr.yml:
```diff
@@ -66,3 +66,35 @@ jobs:
+          --host=aarch64-apple-darwin \
+          --enable-ios=iPhoneOS \
+          --enable-racket=${{ github.workspace}}/host-racket/bin/racket \
+          --enable-scheme=${{ github.workspace}}/host-racket/src/build/cs/c
```
same - space after `github.workspace` for consistency.

pmatos reviewed (changes requested):
I don't have a way to test it, but overall looks good to me.
Also - should this also be part of the push setup, instead of just PR?

Bogdanp commented:
Thanks for the review! I've fixed the spacing and added a job for push as well. I'll test it on my fork (I don't think there's a way to test it here w/o merging, is there?) and then circle back here when it passes.

Bogdanp commented:
The `push` workflow [passes](https://github.com/Bogdanp/racket/runs/1721529887?check_suite_focus=true) so I've squashed the commits.

pmatos reviewed (approved):
lgtm
